update_manifest.py

```diff
@@ -1,29 +1,85 @@
 import hashlib
-import os
+import logging
+import pathlib
+from typing import Optional
 import xml.etree.ElementTree
 
+"""This script requires at least Python 3.7.0 to run."""
 
-def update_manifest():
-    manifest = xml.etree.ElementTree.parse("manifest.xml")
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+
+def update_manifest(version: Optional[str] = None, replace: bool = False):
+    """Update SHA1 hashes and version number for Path of Building's manifest file.
+
+    :param version: Three-part version number following https://semver.org/.
+    :param replace: Whether to overwrite the existing manifest file.
+    :return:
+    """
+    supported_extensions = {".dll", ".jpg", ".lua", ".md", ".png", ".txt"}
+    try:
+        manifest = xml.etree.ElementTree.parse("manifest.xml")
+    except FileNotFoundError:
+        logger.critical(f"Manifest file not found in path '{pathlib.Path().cwd()}'")
+        return
     root = manifest.getroot()
 
     for file in root.iter("File"):
-        path = file.get('name')
-        extension = os.path.splitext(path)[1]
-        if extension not in (".dll", ".lua", ".md", ".txt", ".png", ".jpg"):
-            print(f"Skipping file type {extension}")
+        path = pathlib.Path(file.get("name"))
+        if path.suffix not in supported_extensions:
+            logger.debug(f"Skipping file type {path.suffix}")
             continue
         try:
-            with open(path, 'rb') as f:
-                data = f.read()
-            sha1_hash = hashlib.sha1(data).hexdigest()
-            file.set("sha1", sha1_hash)
-            print(f"Path: {path} hash: {sha1_hash}")
+            data = path.read_bytes()
         except FileNotFoundError:
-            print(f"File not found, skipping {path}")
+            logger.error(f"File not found, skipping {path}")
+            continue
+        sha1_hash = hashlib.sha1(data).hexdigest()
+        file.set("sha1", sha1_hash)
+        logger.info(f"Path: {path} hash: {sha1_hash}")
+    if version:
+        root.find("Version").set("number", version)
+        logger.info(f"Updated to version {version}")
+
+    file_name = "manifest.xml" if replace else "manifest-updated.xml"
+    manifest.write(file_name, encoding="utf-8", xml_declaration=True)
```

Comment:
While still technically spec-compliant, the encoding should be uppercase
```suggestion
    manifest.write(file_name, encoding="UTF-8", xml_declaration=True)
```